### PR TITLE
Soft deprecate WithAuthority on request level

### PR DESCRIPTION
**Pull request by bgavrilMS**
Fixes #

**Changes proposed in this request**
Hide `WithAuthority` methods.

**Testing**

**Performance impact**

**Documentation**
- [ ] All relevant documentation is updated.

### CODE REVIEW

**Review by jmprieur — APPROVED** [review]
LGTM
Thanks @bgavrilMS 

do we want to add an obsolete(false) attribute?

**Review by SameerK-MSFT — APPROVED** [review]

**Review by gladjohn — APPROVED** [review]

**Review by pmaytak — COMMENTED** [review]
What about:
- https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/9af5689b52dbd05c3d23e857cb9f3f2530f46955/src/client/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs#L260
- https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/blob/9af5689b52dbd05c3d23e857cb9f3f2530f46955/src/client/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs#L360

File: src/client/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs [hunk 1]
```diff
@@ -122,6 +123,8 @@ public T WithExtraQueryParameters(string extraQueryParameters)
         }
 
         /// <summary>
+        /// Important: Use WithTenantId or WithTenantIdFromAuthority instead, or WithB2CAuthority for B2C authorities.
```
Comment (pmaytak) [inline src/client/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs:126]: > Use WithTenantId or WithTenantIdFromAuthority instead

Does this include ADFS and DSTS authortities?

**Comment by bgavrilMS** [comment]
> LGTM Thanks @bgavrilMS
> 
> do we want to add an obsolete(false) attribute?

I had a long conversation with Azure SDK folks about this and they advise against it. This is because of diamond dependencies again, which can cause warnings. I think the sceanrio is like this: 

- App references MSAL 4.40, uses an API which is not obsolete
- App also references Azure.Identiy, which references MSAL 4.48, where API is obsolete
- App now gets a warning (and many apps use "report warnings as errors", which breaks the build)

So they recommend just hiding obsolete APIs.

CC: @pmaytak @neha-bhargava @SameerK-MSFT @trwalke @gladjohn

File: src/client/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs (hunk 1 above)
Comment (bgavrilMS) [reply src/client/Microsoft.Identity.Client/ApiConfig/AbstractAcquireTokenParameterBuilder.cs:126]: No, those are not supported scenarios

**Pull request merged**